docker/nginx/nginx.dockerfile

```diff
@@ -1,11 +1,44 @@
-FROM nginx:latest
+# http, https
+ARG SERVER_SCHEME=http
+
+# http, http2
+ARG SERVER_PROTOCOL=http
+
+# NOTE: nginx doesn't support http2 connections to the upstream server
+
+FROM nginx:latest AS base
 
 ENV WORKER_PROCESSES auto
-ENV PROXY_PASS http://tfb-database:5000
+ENV DOWNSTREAM_SCHEME http
+ENV DOWNSTREAM_ADDRESS tfb-database
+ENV DOWNSTREAM_PORT 8080
 
-ADD nginx.conf /etc/nginx/nginx.conf
+ADD testCert.rsa /etc/ssl/private/testCert.rsa
+ADD testCert.crt /etc/ssl/certs/testCert.crt
 
 ADD run.sh /
 RUN chmod +x /run.sh
 
+# Listening to http connections
+FROM base AS scheme-http-http
+# ARG SERVER_SCHEME 
+ADD nginx-http.conf /etc/nginx/nginx.conf
+
+# Listening to https connections
+FROM base AS scheme-https-http
+# ARG SERVER_SCHEME
+ADD nginx-https.conf /etc/nginx/nginx.conf
+
+# Listening to h2 connections
+FROM base AS scheme-https-http2
```

Comment:
It should be HTTP/2 - plaintext HTTP 1.1.

Reply:
This is ${SERVER_SCHEME}-${SERVER_PROTOCOL}
Nothing to do with upstream

Reply:
http-http2 would mean h2c for instance

Reply:
I got it now. Sorry for the misunderstanding.